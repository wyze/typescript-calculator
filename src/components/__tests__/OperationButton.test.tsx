import { css } from 'glamor'
import { h } from 'preact'
import OperationButton from '../OperationButton'
import snapshot from './utils/snapshot'

const props = {
  action: jest.fn(),
  text: '+',
}

const propsWithClassName = {
  ...props,
  className: css({ color: 'rebeccapurple' }),
}

describe('<OperationButton />', () => {
  it('renders', () => snapshot(<OperationButton {...props} />))
  it('renders with className', () => snapshot(<OperationButton {...propsWithClassName} />))
})
