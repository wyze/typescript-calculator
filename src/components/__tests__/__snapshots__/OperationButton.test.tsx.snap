// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OperationButton /> renders 1`] = `
<Button
  action={[Function]}
  className={
    Object {
      "data-css-a5zy9": "",
    }
  }
  text="+"
/>
`;

exports[`<OperationButton /> renders with className 1`] = `
<Button
  action={[Function]}
  className={
    Object {
      "data-css-13btn4m": "",
    }
  }
  text="+"
/>
`;
